Preferences & Config editors

melody-rs opened the pull request:
**Connections**
#43
#45

**Description**
#43 and #45 changed how configuring Luminol worked, but left the configuration editors in an unfinished state. This PR fixes that!
As well as fixing up the config editors, this PR adds support for loading other data formats (but not converting projects, that requires either another out-of-project manager, or changes to the path cache to recognize file extensions when present)

**Testing**
 - (From an old commit where conversion was possible) Load a project, change the format, and reopen it
 - Open the preferences editor and add the path to the `Standard` RTP, then open a project that requires it
 - Edit Game.ini, save, and see the changes
 - Edit Luminol project settings, save, and see changes

<!-- 
Thanks for filing a pull request! The codeowners file will automatically request reviews.

After you get a review and have addressed any comments, please explicitly re-request a review from the
person(s) who reviewed your changes. This will make sure it gets re-added to their review queue - you're not bothering us!
-->

**Checklist**

- [x] Run `cargo fmt`.
- [x] Run `cargo clippy`. If applicable, add:
  - [x] `--target wasm32-unknown-unknown -Z build-std=std,panic_abort`
- [x] Run `cargo build --release`
- [x] If applicable, run `trunk build --release`

melody-rs committed:
Kinda port over preferences editor from old branch

melody-rs committed:
Fix catppuccin visuals

melody-rs committed:
Change theme handling

melody-rs committed:
Fix theme saving

melody-rs committed:
Make use_ron option a combobox

melody-rs committed:
Add format-agnostic handler

melody-rs committed:
Don't pass extension to handler

melody-rs committed:
kinda implement format conversion

melody-rs committed:
Actually get data conversion to work

melody-rs committed:
Change warning message

melody-rs committed:
Use indexmap for rtp paths

melody-rs committed:
Fix some wasm things

melody-rs committed:
Merge remote-tracking branch 'upstream/dev' into preferences

white-axe commented on crates/ui/src/windows/config_window.rs:
Changing the entries in the config window should probably mark the project as modified so that the unsaved changes modal pops up when closing the project without saving.

Also, the scripts path setting doesn't seem to persist after saving the project. At least in the config window.

melody-rs replied:
The scripts paths setting appears not to change when reloading because if Luminol fails to open the scripts file, it'll try to guess the scripts path and set the setting to the guess if it succeeds

I'll adjust that behavior

melody-rs replied:
Creating new files in the path cache appears not to be working currently
![image](https://github.com/user-attachments/assets/d6815155-f8d9-4904-ae71-6261b135a561)

melody-rs replied:
I've done my best to fix this, (editing scripts path will mark scripts as modified so they save too, project displays errors when attempting to load scripts) but this seems to be an underlying issue with the path cache

white-axe replied:
Okay, I'll investigate this later

melody-rs committed:
Make config window set modified flag

melody-rs committed:
Change how scripts_path functions
Produces a toast when failing to load scripts, and when modified marks scripts as modified
